prelude/__internal__/IO.mad

```diff
@@ -2,14 +2,139 @@ import type { Wish } from "Wish"
 import { Wish } from "Wish"
 import IOError from "IOError"
 
+import { mapWithIndex } from "List"
+import { equals, when } from "Function"
+import { split, join } from "String"
+
+END_COLOR = "\x1b[0m"
+
+colorize :: String -> String -> String
+export colorize = (color, v) => color ++ v ++ END_COLOR
+
+uncolorize :: String -> String
+export uncolorize = (v) => END_COLOR ++ v
+
+// via https://en.wikipedia.org/wiki/ANSI_escape_code
+
+export ansi = {
+  FGBlack: "30",
+  FGRed: "31",
+  FGGreen: "32",
+  FGYellow: "33",
+  FGBlue: "34",
+  FGMagenta: "35",
+  FGCyan: "36",
+  FGWhite: "37",
+
+  FGBrightBlack: "90",
+  FGBrightRed: "91",
+  FGBrightGreen: "92",
+  FGBrightYellow: "93",
+  FGBrightBlue: "94",
+  FGBrightMagenta: "95",
+  FGBrightCyan: "96",
+  FGBrightWhite: "97",
+
+  BGBlack: "40",
+  BGRed: "41",
+  BGGreen: "42",
+  BGYellow: "43",
+  BGBlue: "44",
+  BGMagenta: "45",
+  BGCyan: "46",
+  BGWhite: "47",
+
+  BGBrightBlack: "100",
+  BGBrightRed: "101",
+  BGBrightGreen: "102",
+  BGBrightYellow: "103",
+  BGBrightBlue: "104",
+  BGBrightMagenta: "105",
+  BGBrightCyan: "106",
+  BGBrightWhite: "107",
+
+  FormatUnderline: "4",
+  FormatNoUnderline: "24",
+  FormatBold: "1",
+  FormatNoBold: "21",
+
+  FormatInvert: "7",
+}
+
+ansiColor :: List String -> String -> String
+export ansiColor = (parts, str) => colorize(
+  `\x1b[${join(";", parts)}m`,
+  str
+)
 
-withColor :: String -> String -> String
-withColor = (color, v) => color ++ v ++"\x1b[0m"
+export text = {
```

Comment:
Also potentially these could be cased differently like:
```madlib
export Text = {
  BLACK: ...,
  RED: ...,
   ...
}
```

So that we write:
```madlib
Text.DARK
```

Reply:
I don't feel super strongly but in my mind capitalized things tend to be constructors / types -- I think in practice you'll end up using a fair amount of initial boilerplate -- the `text` object is more for standardization of patterns, in a recent file I had something like:
```
import {colortrace, text} from "@/Log"

red = colortrace(text.red)
red("cool shit", 420)
```

Reply:
Yeah I get it, my idea of it was that it behave like namespaced constants. Happy to go with that and define some rules at some point!